core/src/conn.rs

```diff
@@ -16,20 +21,44 @@ pub struct MongoConnection {
     pub operation_timeout: Option<Duration>,
 }
 
+const MONGODB_ODBC_DRIVER: &str = "mongo_odbc_driver";
+
 impl MongoConnection {
-    // Creates a new MongoConnection with the given settings and execute a dummy aggregation to
-    // validate it works.
-    // The operation will timeout if it takes more than loginTimeout seconds.
+    // Creates a new MongoConnection with the given settings and lists databases to make sure the
+    // connection is legit.
+    //
+    // The operation will timeout if it takes more than loginTimeout seconds. This timeout is
+    // delegated to the mongo rust driver.
+    //
     // The initial current database if provided should come from SQL_ATTR_CURRENT_CATALOG
     // and will take precedence over the database setting specified in the uri if any.
-    // The initial operation time if provided should come from  and will take precedence over the
+    // The initial operation time if provided should come from and will take precedence over the
     // setting specified in the uri if any.
     pub fn connect(
-        _uri: &str,
-        _current_db: Option<&str>,
-        _operation_timeout: Option<i32>,
-        _login_timeout: Option<i32>,
+        mongo_uri: &str,
+        auth_src: &str,
+        current_db: Option<&str>,
+        operation_timeout: Option<u32>,
+        login_timeout: Option<u32>,
+        application_name: Option<&str>,
     ) -> Result<Self> {
-        unimplemented!()
+        let mut client_options = ClientOptions::parse(mongo_uri)?;
+        client_options.connect_timeout = login_timeout.map(|to| Duration::new(to as u64, 0));
+        // set application name, note that users can set their own application name, or we default
+        // to mongo-odbc-driver.
+        client_options.app_name = application_name
+            .map(String::from)
+            .or_else(|| Some(MONGODB_ODBC_DRIVER.to_string()));
+        let client = Client::with_options(client_options)?;
+        // run the "ping" command on the `auth_src` database. We assume this requires the
+        // fewest permissions of anything we can do to verify a connection.
+        client
+            .database(auth_src)
+            .run_command(doc! {"ping": 1}, None)?;
```

Comment:
This is different from the JDBC driver where we are doing "SELECT 1 from DUAL". Is there a reason? Is one better than the other and in this case, should we update the JDBC driver at some point?

Reply:
I think we should change the JDBC driver.

Reply:
Sounds good. I create [SQL-998](https://jira.mongodb.org/browse/SQL-998) for this.